### PR TITLE
Visualize objective plot as dashed lines

### LINKED ISSUE #77: Objective values plots, track geo realizations
During the user visualization meeting, there was a discussion regarding the objective plots, some users wanted labeled realizations to track geo realizations, or potentially drawn lines instead of scattered points, potentially have geo realization filter.

Some discussion regarding what is the right approach is still needed.

### PR DESCRIPTION
**Pull request by mortalisk**
Resolves #77

Draw each line with different colour. When several functions are
drawn, each function will be of different colour as before.
Also add a feature where hovering over a point will draw the
corresponding realization with a thicker full line, in order to be
able to track individual realizations in the case of several functions.

### CODE REVIEW

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 1]
```diff
@@ -181,10 +181,11 @@ def set_button_enabled_state(filter_realizations):
                 Input(self.radio_id_mode, "value"),
                 Input(self.realization_filter_check_id, "value"),
                 Input(self.realization_filter_input_id, "value"),
+                Input(self.graph_id, "hoverData"),
```
Comment (DanSava) [inline everviz/plugins/objectives_plot/objectives_plot.py:184]: I would replace `hoverData` this with `clickData`

Using hoverData causes some strange flicker effect and hides the values of the points user is hovering over

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 2]
```diff
@@ -181,10 +181,11 @@ def set_button_enabled_state(filter_realizations):
...
             ],
         )
         def update_graph(
-            func_list, radio_value, plot_mode, realizations_check, realizations_input
+            func_list, radio_value, plot_mode, realizations_check, realizations_input, hoverData
```
Comment (DanSava) [inline everviz/plugins/objectives_plot/objectives_plot.py:188]: `hoverData` should not be camel case maybe hover_data? or click_data if you go with the replacement on line 184.

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 3]
```diff
@@ -247,26 +248,44 @@ def update_graph(
...
+                        if radio_value == "Normalized":
+                            y_values = y_values * line["norm"]
+                        elif radio_value == "Weighted + Normalized":
+                            y_values = y_values * line["norm"] * line["weight"]
+                        line_style = {"dash": "dash"}
+                        if len(func_list) > 1:
+                            line_style["color"] = color
+
+                        if hoverData is not None:
+                            hover_real = hoverData["points"][0]["customdata"]
```
Comment (DanSava) [inline everviz/plugins/objectives_plot/objectives_plot.py:266]: This results in an error when the user hovers away from a point 
```
/everest/lib/python3.6/site-packages/everviz/plugins/objectives_plot/objectives_plot.py", line 266, in update_graph
    hover_real = hoverData["points"][0]["customdata"]
KeyError: 'customdata'
```
I suggest using 

`hover_real = hoverData["points"][0].get("customdata", None)`

or better yet
`clicked_realization = click_data["points"][0].get("customdata", None)`

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 4]
```diff
@@ -247,26 +248,44 @@ def update_graph(
...
+                                line_style["width"] = 4
+                                del line_style["dash"]
+
+                        traces.append(
+                            go.Scatter(
+                                y=y_values,
+                                x=line["batch"],
+                                mode="lines+markers ",
+                                marker={"size": 10},
+                                name=f"{key}({real})",
```
Comment (DanSava) [inline everviz/plugins/objectives_plot/objectives_plot.py:277]: I think the name would look better like {key}_{real} or {key}_r_{real}

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 5]
```diff
@@ -215,58 +221,98 @@ def update_graph(
...
-                    traces.append(
-                        go.Scatter(
-                            y=y_values,
-                            x=func_data["batch"],
-                            mode=plot_mode,
-                            marker={"size": 10},
-                            name=key,
-                            showlegend=True,
+                    traces.extend(
+                        ObjectivesPlot.function_traces(
```
Comment (DanSava) [inline everviz/plugins/objectives_plot/objectives_plot.py:230]: Same question as above

**Comment by DanSava** [comment]
Looks good :+1:  just had one question

File: everviz/plugins/objectives_plot/objectives_plot.py [hunk 6]
```diff
@@ -215,58 +221,98 @@ def update_graph(
...
-                                y=func_data["Mean"],
-                                x=func_data["batch"],
-                                mode=plot_mode,
-                                marker={"size": 10},
-                                line={"color": color},
-                                name=key,
-                                showlegend=True,
-                            ),
-                        ]
+                        ObjectivesPlot.statistics_traces(
```
[the remark this answers is not in the document]
Comment (mortalisk) [reply everviz/plugins/objectives_plot/objectives_plot.py:224]: No reason. I can make it use self if you prefer.

File: everviz/plugins/objectives_plot/objectives_plot.py (hunk 5 above)
Comment (mortalisk) [reply everviz/plugins/objectives_plot/objectives_plot.py:230]: Same answer as above

File: everviz/plugins/objectives_plot/objectives_plot.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (mortalisk) [reply everviz/plugins/objectives_plot/objectives_plot.py]: Well, now that I changed it, it seems I did it right in the first place. Pylint complains:
`R0201: Method could be a function (no-self-use)`
I can either change it back, or move the method definitions outside the class scope altogether. Which do you prefer?

[the remark this answers is not in the document]
Comment (DanSava) [reply everviz/plugins/objectives_plot/objectives_plot.py]: I would prefer moving them outside the class scope if that is not too much trouble.

**Pull request merged**